File: src/main/java/no/nav/nare/core/evaluation/Resultat.java
package no.nav.nare.core.evaluation;


public enum Resultat {

    JA(-1, "NEI"),
    NEI(1, "JA");

    private final int weight;
    private String inverse;


    Resultat(int weight, String inverse) {
        this.weight = weight;
        this.inverse = inverse;
    }



    public Resultat and(Resultat resultat) {
        return (this.weight > resultat.weight) ? this : resultat;
    }

    public Resultat or(Resultat resultat) {
        return (this.weight < resultat.weight) ? this : resultat;
    }

    public Resultat not(){
        return Resultat.valueOf(this.inverse);

    }
}

File: src/main/kotlin/no/nav/knare/core/evaluations/Resultat.kt
package no.nav.knare.core.evaluations

enum class Resultat {
    JA {
        override infix fun and(other: Resultat): Resultat = other
        override infix fun or(other: Resultat): Resultat = JA
        override fun not(): Resultat = NEI
    },

    NEI {
        override infix fun and(other: Resultat): Resultat = NEI
        override infix fun or(other: Resultat): Resultat = other
        override fun not(): Resultat = JA
    },

    KANSKJE {
        override infix fun and(other: Resultat): Resultat = if (other == JA) KANSKJE else other
        override infix fun or(other: Resultat): Resultat = if (other == NEI) KANSKJE else other
        override fun not(): Resultat = KANSKJE
    };

    abstract infix fun and(other: Resultat): Resultat
    abstract infix fun or(other: Resultat): Resultat
    abstract fun not(): Resultat
}